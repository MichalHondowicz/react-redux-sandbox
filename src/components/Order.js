import React from "react";
import OrderItem from "./OrderItem";

class Order extends React.Component {
  render() {
    const orderList = this.props.order.map(book => {
      return (
        <OrderItem book={book} removeFromOrder={this.props.removeFromOrder} />
      );
    });
    const orderSummary = this.props.order
      .map(book => Number(book.price))
      .reduce((sum, next) => sum + next, 0);

    return (
      <div className="inventory col-md-6">
        <h2>Cart: </h2>
        {orderList}
        {orderSummary}
      </div>
    );
  }
}

export default Order;
