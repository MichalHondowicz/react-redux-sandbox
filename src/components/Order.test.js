import ReactDOM from "react-dom";
import Order from "./Order";
import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'


configure({adapter: new Adapter()});

describe('Order test', () => {
    it('Order renders with no problems ', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Order/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Inventory rendered', () => {
        const wrapper = shallow(<Order/>);
        expect(wrapper.find('div').text()).toBe('Order');
    });

    it('Snapshot ok', () => {
        const wrapper = shallow(<Order/>);
        expect(wrapper).toMatchSnapshot();
    })
});