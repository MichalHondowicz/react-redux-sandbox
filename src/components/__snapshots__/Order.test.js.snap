// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order test Order Snapshot ok 1`] = `ShallowWrapper {}`;
